cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/messaging/aws/sqs/SqsTypedMessageQueue.java

```diff
@@ -63,4 +72,40 @@ protected TypedMessage toMessage(final com.amazonaws.services.sqs.model.Message
         }
     }
 
+    @Override
+    public List<TypedMessage> receive() throws MessageReceiveException {
+        final List<TypedMessage> receivedMessages = super.receive();
+        logReceivedMessages(receivedMessages);
+        return receivedMessages;
+    }
+
+    @Override
+    public List<TypedMessage> receive(final int waitTimeSeconds, final int maxMessages) throws MessageReceiveException {
+        final List<TypedMessage> receivedMessages = super.receive(waitTimeSeconds, maxMessages);
+        logReceivedMessages(receivedMessages);
+        return receivedMessages;
+    }
+
+    @Override
+    public void send(final TypedMessage message) throws MessageSendException {
+        logSendMessage(message);
```

Comment:
Similar to above, we may need to control which queues are subject to logging when sending messages. I'm not sure on this though.